snap-geometry-edits/src/main/java/com/esri/arcgismaps/sample/snapgeometryedits/components/MapViewModel.kt

```diff
@@ -0,0 +1,247 @@
+        }
+    }
+
+    /**
+     * Synchronises the snap source collection with the map's operational layers, sets the bottom
+     * sheet UI, and shows it to configure snapping.
+     */
+    fun showBottomSheet() {
+        if (geometryEditor.snapSettings.sourceSettings.isEmpty()) {
+            // sync the snap source collection
+            geometryEditor.snapSettings.syncSourceSettings()
+            // initialise the snap source lists used for the bottom sheet
+            geometryEditor.snapSettings.sourceSettings.forEach { snapSource ->
+                snapSourceCheckedState.add(snapSource.isEnabled)
+            }
+            _snapSourceSettingsList.value = geometryEditor.snapSettings.sourceSettings
+        }
+        isBottomSheetVisible.value = true
+    }
+
+    /**
+     * Toggles snapping using the [checkedValue] from the bottom sheet.
+     */
+    fun snappingEnabledStatus(checkedValue: Boolean) {
+        snappingCheckedState.value = checkedValue
+        geometryEditor.snapSettings.isEnabled = snappingCheckedState.value
+    }
+
+    /**
+     * Toggles snapping for the snap source at [index] using the [checkedValue] from the
+     * BottomSheet.
+     */
+    fun sourceEnabledStatus(checkedValue: Boolean, index: Int) {
+        snapSourceCheckedState[index] = checkedValue
+        geometryEditor.snapSettings.sourceSettings[index].isEnabled = snapSourceCheckedState[index]
+    }
+
+    /**
+     * Hides the bottom sheet.
+     */
+    fun dismissBottomSheet() {
+        isBottomSheetVisible.value = false
+    }
+
+    /**
+     * Starts the GeometryEditor using the selected [GeometryType].
+     */
+    fun startEditor(selectedGeometry: GeometryType) {
+        if (!geometryEditor.isStarted.value) {
+            geometryEditor.start(selectedGeometry)
+            toggleButtons()
+        }
+    }
+
+    /**
+     * Stops the GeometryEditor and updates the identified graphic or calls [createGraphic].
+     */
+    fun stopEditor() {
+        if (identifiedGraphic.geometry != null) {
+            identifiedGraphic.geometry = geometryEditor.stop()
+            identifiedGraphic.isSelected = false
+        } else {
+            if (geometryEditor.isStarted.value) {
+                createGraphic()
+            }
+        }
+        toggleButtons()
+    }
+
+    /**
+     * Creates a graphic from the geometry and add it to the GraphicsOverlay.
+     */
+    private fun createGraphic() {
+        val geometry = geometryEditor.stop()
+        val graphic = Graphic(geometry)
+
+        when (geometry!!) {
+            is Point -> graphic.symbol = GeometryEditorStyle().vertexSymbol
+            is Multipoint -> graphic.symbol = GeometryEditorStyle().vertexSymbol
+            is Polyline -> graphic.symbol = GeometryEditorStyle().lineSymbol
+            is Polygon -> graphic.symbol = GeometryEditorStyle().fillSymbol
+            is Envelope -> graphic.symbol = GeometryEditorStyle().lineSymbol
```

Comment:
Geometry from a GeometryEditor will not be an `Envelope`.

```suggestion
            else -> {}
```